cql3/statements/create_index_statement.cc

```diff
@@ -343,9 +343,11 @@ std::optional<create_index_statement::base_schema_with_new_index> create_index_s
     }
```

Comment:
Pls split the commit into smaller ones. For instance, this one can be split into: 
- allowing to create custom index with `dummy-vector-backend` class
- adjusting describe statement
- ... (maybe others)

Moreover, each commit should have a title following pattern: `path/to/module: what the commit is doing` and a description which explains motivation and what exactly the commit is doing. (some example commit: https://github.com/scylladb/scylladb/pull/21471/commits/de7b58e8d47874d44d8b2de0be99936e580f2519)